firmware/library/L3_HAL/sddriver.hpp

```diff
@@ -0,0 +1,239 @@
+#pragma once
+
+#include <cstdint>
+#include "L1_Drivers/gpio.hpp"
+#include "L1_Drivers/ssp.hpp"
+#include "L2_Utilities/debug_print.hpp"
+
+// Relevant Datasheet: https://www.sdcard.org/downloads/pls/pdf/index.php?p=Part1_Physical_Layer_Simplified_Specification_Ver6.00.jpg&f=Part1_Physical_Layer_Simplified_Specification_Ver6.00.pdf&e=EN_SS1
+class SdInterface
+{
+ public:
+    // // @constructor     SdInterface()
+    // constexpr SdInterface() = 0;
+
+    // @destructor      ~SdInterface
+    virtual ~SdInterface() {};
```

Comment:
There is no need for a virtual, override-able virtual destructor for an interface.